### ISSUE TITLE
Error again...

### ISSUE THREAD

**Issue by Jayshua83** [loki79uk/FS22_LumberJack#25]
Getting this error... It is a different "line" number? than before, not sure if that matters!

2023-08-25 20:36 Error: Running LUA method 'update'.
dataS/scripts/player/Player.lua(3110) : attempt to perform arithmetic on local 'inputValue' (a nil value)

#15

**Comment by Jayshua83** [comment]
Frustrating times trying to figure out why this is happening and what causes it! :|

**Comment by loki79uk** [comment]
I've made a new TEST version that should print out some call stacks when this happens and maybe help to find the cause.  
[FS22_LumberJack.zip](https://github.com/loki79uk/FS22_LumberJack/files/12444917/FS22_LumberJack.zip)

**Comment by loki79uk** [comment]
I've made a new test version that should print out some call stacks when this happens, and hopefully help to identify what is going on..  [FS22_LumberJack.zip](https://github.com/loki79uk/FS22_LumberJack/files/12444928/FS22_LumberJack.zip)

**Comment by bazillusX** [comment]
Hello, used the file above, since we got the same error (https://github.com/loki79uk/FS22_LumberJack/issues/25#issue-1867827245). Playing with me as a host and a mate as client, while i got the log message when the client tries to grab something. In this case it was a truck, had a error before with a sports bag, but forgot to save the error and we were not able to reproduce it with the bag.

> 2023-12-16 23:00 rotateObject: inputValue NIL
> 2023-12-16 23:00 LUA call stack:
>   =C:/Users/Administrator/Documents/My Games/FarmingSimulator2022/mods/FS22_LumberJack/LumberJack.lua (872) : printCallstack
>   dataS/scripts/player/Player.lua (931) : rotateObject
>   scripts/PlayerExtension.lua (37) : oldPlayerUpdate
>   dataS/scripts/network/NetworkNode.lua (163) : update
>   dataS/scripts/network/Server.lua (62) : updateActiveObjects
>   dataS/scripts/BaseMission.lua (1148) : update
>   dataS/scripts/FSBaseMission.lua (1983) : update
>   dataS/scripts/missions/mission00.lua (577) : update
>   dataS/scripts/main.lua (1289) : update
> 2023-12-16 23:00 rotateObject: inputValue NIL
> 2023-12-16 23:00 LUA call stack:
>   =C:/Users/Administrator/Documents/My Games/FarmingSimulator2022/mods/FS22_LumberJack/LumberJack.lua (872) : printCallstack
>   dataS/scripts/player/Player.lua (931) : rotateObject
>   scripts/PlayerExtension.lua (37) : oldPlayerUpdate
>   dataS/scripts/network/NetworkNode.lua (163) : update
>   dataS/scripts/network/Server.lua (62) : updateActiveObjects
>   dataS/scripts/BaseMission.lua (1148) : update
>   dataS/scripts/FSBaseMission.lua (1983) : update
>   dataS/scripts/missions/mission00.lua (577) : update
>   dataS/scripts/main.lua (1289) : update

**Comment by loki79uk** [comment]
@bazillusX Was there any error in the game, or just this printed in the log?

I added this function in the version above (and some others):
```lua
Player.rotateObject = Utils.overwrittenFunction(Player.rotateObject,
function(self, superFunc, inputValue, axisX, axisY, axisZ)
	if self and superFunc and inputValue then
		return superFunc(self, inputValue, axisX, axisY, axisZ)
	else
		print("rotateObject: inputValue NIL")
		printCallstack()
	end
end)
```
so that is where the message and print callstack is coming from.

**Comment by loki79uk** [comment]
@bazillusX  Or if you could send me any log file from that savegame, I will look for mods that might be good candidates.

**Comment by loki79uk** [comment]
Thanks to FSG for identifying the mod, it appears to be `FS22_MapObjectsHider`.  The rotateObject issue may happen anyway, and this mod is just in the call stack.

I don't have time to investigate properly now, but if you could try without that mod for a while it would be helpful to confirm/eliminate that as a possibility.  Equally if anyone else with the same issue could confirm if they are using this mod or not, it would help a lot!

**Comment by bazillusX** [comment]
> @bazillusX Was there any error in the game, or just this printed in the log?

Basically just the printed log error. We had some hard crashes (endless callstack) while i was playing around on AutoDrive editing courses when we had the first crashes. My mate was simultaneously grabbing bales which were glued to his hands, but this could also relate to the heavy log printing. 

> So I think we are looking for a mod with a file `scripts/PlayerExtension.lua`, that has a line 37 calling `oldPlayerUpdate`.

I tried. Searched for the phrase, the file and snippets of them. No luck - maybe DLC?


> @bazillusX Or if you could send me any log file from that savegame, I will look for mods that might be good candidates.

https://pastebin.com/q25aMbV1 List of the mods for the current session. No changes to the last one. If you need the entire file let me know.

**Comment by loki79uk** [comment]
> > So I think we are looking for a mod with a file `scripts/PlayerExtension.lua`, that has a line 37 calling `oldPlayerUpdate`.
> 
> I tried. Searched for the phrase, the file and snippets of them. No luck - maybe DLC?

You are right, it is from the new DLC.  I've deleted my previous post so that no one wastes time checking!

**Comment by FirenzeIT** [comment]
Hello @loki79uk , please add a support compatiblity with mod easyDevelopement, if possible, thanks

**Comment by loki79uk** [comment]
> Hello @loki79uk , please add a support compatiblity with mod easyDevelopement, if possible, thanks

Do you mean where the super strength option is disabled in EasyDev?  That is a feature of EasyDev that was added to support lumberjack, since my implemention is different to that of the base game.  Mine is designed so that you can set different strength values for individual players in multiplayer games.
